src/components/YourFeed.tsx

```diff
@@ -217,15 +218,23 @@ export class YourFeed extends React.PureComponent<DispatchProps & StateProps, Yo
     private onFollowPressed = async (author: Author) => {
         const followedFeed = this.props.feeds.find(feed => feed.feedUrl === author.uri);
         if (followedFeed != null) {
-            const confirmUnfollow = await AreYouSureDialog.show('Are you sure you want to unfollow?');
-            if (confirmUnfollow) {
-                this.props.onUnfollowFeed(followedFeed);
-            }
+            await this.unfollowFeed(followedFeed);
         } else {
-            const knownFeed = this.props.knownFeeds.find(feed => feed.feedUrl === author.uri);
-            if (knownFeed != null) {
-                this.props.onAddFeed(knownFeed);
-            }
+            this.followFeed(author);
+        }
+    }
+
+    private async unfollowFeed(feed: Feed) {
+        const confirmUnfollow = await AreYouSureDialog.show('Are you sure you want to unfollow?');
+        if (confirmUnfollow) {
+            this.props.onUnfollowFeed(feed);
+        }
+    }
+
+    private followFeed(author: Author) {
+        const knownFeed = this.props.knownFeeds.find(feed => feed.feedUrl === author.uri);
+        if (knownFeed != null) {
+            this.props.onAddFeed(knownFeed);
```

Comment:
The logic is still incorrect here: `knownFeeds` is coming from `state.feeds`, so if you find a known feed, you will add it to `state.feeds` therefore it will be added twice.

Reply:
it is not, check the reducer. Maybe the callback should be renamed but not sure to what